cpp/src/parquet/column_reader.cc

```diff
@@ -1762,7 +1763,7 @@ class TypedRecordReader : public TypedColumnReaderImpl<DType>,
   }
 
   // Return number of logical records read
-  int64_t ReadRecordData(int64_t num_records) {
+  int64_t ReadRecordData(int64_t num_records, bool read_dense_for_nullable) {
```

Comment:
If `read_dense_for_nullable` is set to true but the column has null values, it seems that we haven't checked it and the reader may fail with random error or read wrong values.